tests/backup/backup_helper.go

```diff
@@ -2446,21 +2453,19 @@ func deleteJobAndWait(job batchv1.Job) error {
 
 func ValidateAllPodsInPxBackupNamespace() error {
 	pxBackupNamespace, err := backup.GetPxBackupNamespace()
-	allPods, err := core.Instance().GetPods(pxBackupNamespace, nil)
-	for _, pod := range allPods.Items {
-		if strings.Contains(pod.Name, pxCentralPostInstallHookJobName) ||
-			strings.Contains(pod.Name, quickMaintenancePod) ||
-			strings.Contains(pod.Name, fullMaintenancePod) {
-			continue
-		}
-		log.Infof("Checking status for pod - %s", pod.GetName())
-		err = core.Instance().ValidatePod(&pod, 5*time.Minute, 30*time.Second)
-		if err != nil {
-			// Collect mongoDB logs right after the command
-			ginkgoTest := CurrentGinkgoTestDescription()
-			testCaseName := fmt.Sprintf("%s-error", ginkgoTest.FullTestText)
-			CollectMongoDBLogs(testCaseName)
-			return err
+	log.FailOnError(err, "Getting px-backup namespace")
+	for _, label := range backupPodLabels {
+		allPods, err := core.Instance().GetPods(pxBackupNamespace, label)
+		for _, pod := range allPods.Items {
+			log.Infof("Checking status for pod - %s", pod.GetName())
+			err = core.Instance().ValidatePod(&pod, 5*time.Minute, 30*time.Second)
+			if err != nil {
+				// Collect mongoDB logs right after the command
+				ginkgoTest := CurrentGinkgoTestDescription()
+				testCaseName := fmt.Sprintf("%s-error", ginkgoTest.FullTestText)
+				CollectMongoDBLogs(testCaseName)
```

Comment:
can we collect log for any of the pod which fails validation here, 
if it's out of scope of this PR. we can consider in seperate ticket.

Reply:
done